File: LABORATORIOS/LABORATORIO3/Ejercicio5/Java/NumerosMultiplosTres.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class NumerosMultiplosTres {
    public static int generarNumeroMultiplosTres(int x) {
        List<Integer> digitos = new ArrayList<>();
        while (x != 0) {
            digitos.add(x % 10);
            x = x / 10;
        }

        List<Integer> multiplosTres = new ArrayList<>();
        for (int i = 0; i < digitos.size(); i++) {
            int digito = digitos.get(i);
            if (digito % 3 == 0) {
                multiplosTres.add(digito);
            }
        }

        int y = 0;
        int multiplicador = 1;
        for (int i = 0; i < multiplosTres.size(); i++) {
            y += multiplosTres.get(i) * multiplicador;
            multiplicador *= 10;
        }

        return y;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese un número: ");
        int x = scanner.nextInt();
        int y = generarNumeroMultiplosTres(x);
        System.out.println("El número con los dígitos múltiplos de 3 es: " + y);
    }
}

File: LABORATORIOS/LABORATORIO3/Ejercicio5/Python/NumerosMultiplosTres.py
def generar_numero_multiplos_tres(x):
    digitos = [int(d) for d in str(x)]
    multiplos_tres = [d for d in digitos if d % 3 == 0]
    y = int("".join(str(d) for d in multiplos_tres))
    return y

x = int(input("Ingrese un número: "))
y = generar_numero_multiplos_tres(x)
print("El número con los dígitos múltiplos de 3 es:", y)